### DISCUSSION TITLE
prepend characters to Panel

### DISCUSSION THREAD

**Discussion by nschloe** [Textualize/rich#1701] [Q&A] [answered]
[It is known that vertical alignment doesn't work yet in Rich.](https://github.com/willmcgugan/rich/issues/1590) As a workaround, how can I prepend newlines before `Panel` in the following example?
```python
from rich.console import Console
from rich.table import Table
from rich.panel import Panel

console = Console()

table = Table(show_header=False)
table.add_row("a\nb\nc\nd\ne\nf\ng\nh\ni\nj", Panel("test panel"))

console.print(table)
```
![screenshot](https://user-images.githubusercontent.com/181628/142069718-d7d51b38-1b8e-402d-b9f1-04fae3a1a054.png)

**Comment by willmcgugan** [comment, marked as answer]
Vertical alignment does work, but not yet in Table rows.

You can use the Padding class to insert lines before a renderable.

**Reply by nschloe** [reply]
Padding inserts space inside the box. I'm looking for something outside of it, like `margin`.

**Comment by willmcgugan** [comment]
Padding works for both. You can put the Panel inside Padding, or Padding inside Panel. Rich doesn’t care which.

**Reply by willmcgugan** [reply]
https://rich.readthedocs.io/en/stable/padding.html

**Reply by nschloe** [reply]
Yup, that works. Thanks!
